gatsby-config.js

```diff
@@ -145,6 +145,14 @@ const plugins = [
       excludePattern: /\/doc\/cml/
     }
   },
+  {
+    resolve: `gatsby-plugin-algolia`,
```

Comment:
Q: what does it exactly do? send data to Algolia during the build-time? or just helps to render JS?

Reply:
Yes, exactly this helps to send data to Algolia for indexing at build-time. No, this doesn't help to render. We have another utility package called `react-instantsearch-dom` by the Algolia team for rendering.

Reply:
thanks, Thomas, it's clear now.

Q: how much does it slow down the build? how will it work with the preview apps if they share the same index or if there is a way to do subindex? (for local testing, for preview apps, etc?)

Reply:
It's around 2-3 seconds. I just checked on preview apps build.
<img width="429" alt="Screen Shot 2022-01-24 at 10 09 44" src="https://user-images.githubusercontent.com/20840228/150721882-d231dc73-2eff-4010-94d7-524487e30362.png">
For indexing, we can set different environment variables for each environment. 
<img width="1688" alt="Screen Shot 2022-01-24 at 10 13 47" src="https://user-images.githubusercontent.com/20840228/150722218-89e7c0d0-f5ec-4963-9e4a-714f810b413a.png">
I have set `stag_blogs` as indexing for preview apps. 
We can only set one environment variable for preview apps, so the same index will be used for all preview apps. I don't that will be a problem.
Similarly, we can set a variable for production as well. For eg: `prod_blogs` should separate it from others.

Reply:
I have now updated the keys on preview apps with the Algolia application `DVC Blog`(One that you have invited me to). So, newer builds should start using those.

Reply:
sounds good, Thomas.

Still heve mixed feelings about sending every blog text to Algolia every time we rebuild 🤔 

@rogermparent @julieg18 what's your take?

Reply:
I think running a full update every time is fine for a first iteration, particularly if it's only a few seconds with the amount we have now, but we can also consider another iteration on top of this that takes advantage of the [partial updates feature](https://www.gatsbyjs.com/plugins/gatsby-plugin-algolia/#partial-updates) of `gatsby-plugin-algolia`. We'll have to add our own `modified` field providing either a boolean or datetime, but I'm sure we can figure that out.

Reply:
Agreed with @rogermparent! We can use this for a first iteration and open a issue for improving this at a later date.

Reply:
>We'll have to add our own modified field providing either a boolean or datetime, but I'm sure we can figure that out.

Thanks, for mentioning that out. Yeah, that would be better and more scalable. And, for `modified`, I found we can't just use the file modified datetime as that changes when the file is on different places(local, server) instead we could use the git method to get the modified datetime. [Stackoverflow](https://stackoverflow.com/a/66118168).
I would check more on that.